app/src/main/java/com/example/solom/hotel_csv_app/MainActivity.java

```diff
@@ -18,6 +25,21 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        ListView listView = findViewById(R.id.listview);
+        itemArrayAdapter = new ItemArrayAdapter(getApplicationContext(), R.layout.single_list_item);
+
+        Parcelable state = listView.onSaveInstanceState();
+        listView.setAdapter(itemArrayAdapter);
+        listView.onRestoreInstanceState(state);
+
+        InputStream inputStream = getResources().openRawResource(R.raw.stats);
```

Comment:
Performance can be improved by moving the code that parses the CSV file to a new Thread